### PR TITLE
Rename opendistro rest apis

### PR DESCRIPTION
**Pull request by eirsep**
### Description
Renames rest endoints with _opensearch path prefix

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public static Request buildHttpRequest(SubmitAsynchronousSearchRequest submitAsy
         SearchRequest searchRequest = submitAsynchronousSearchRequest.getSearchRequest();
         Request request = new Request(HttpPost.METHOD_NAME,
                 /*trim first backslash*/
-                endpoint(searchRequest.indices(), AsynchronousSearchPlugin.BASE_URI.substring(1)));
+                endpoint(searchRequest.indices(), AsynchronousSearchPlugin.OPENSEARCH_BASE_URI.substring(1)));
```
Comment (Bukhtawar) [inline src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java:76]: should we do a random between the endpoints to ensure both of them work in the meantime

Comment (dblock) [reply src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java:76]: I think you want a couple of predictable tests for backward compatibility instead.

File: src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java [hunk 2]
```diff
@@ -85,6 +85,7 @@
 
     public static final String OPEN_DISTRO_ASYNC_SEARCH_GENERIC_THREAD_POOL_NAME = "opendistro_asynchronous_search_generic";
     public static final String BASE_URI = "/_opendistro/_asynchronous_search";
```
Comment (dblock) [inline src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java:87]: I would rename this to `OPENDISTRO_BASE_URI` for consistency and avoiding confusion on which one to use for the future.

**Review by dblock — COMMENTED** [review]
This works, but are we sure we don't want to 301 from the old path?

File: src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java [hunk 3]
```diff
@@ -71,9 +72,11 @@
     public static Request buildHttpRequest(SubmitAsynchronousSearchRequest submitAsynchronousSearchRequest) throws IOException {
 
         SearchRequest searchRequest = submitAsynchronousSearchRequest.getSearchRequest();
+        String baseUri = Randomness.get().nextBoolean() ? AsynchronousSearchPlugin.BASE_URI
```
Comment (Bukhtawar) [inline src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java:75]: Do we just need to change it here?

Comment (eirsep) [reply src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/utils/RestTestUtils.java]: added this change in delete and get apis. thanks for catching that!

File: src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java [hunk 4]
```diff
@@ -84,7 +84,8 @@
 public class AsynchronousSearchPlugin extends Plugin implements ActionPlugin, SystemIndexPlugin {
 
     public static final String OPEN_DISTRO_ASYNC_SEARCH_GENERIC_THREAD_POOL_NAME = "opendistro_asynchronous_search_generic";
-    public static final String BASE_URI = "/_opendistro/_asynchronous_search";
+    public static final String LEGACY_BASE_URI = "/_opendistro/_asynchronous_search";
```
Comment (dblock) [inline src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java:87]: `LEGACY_OPENDISTRO_BASE_URI`, who knows how may legacies we'll introduce ... that's also what we have in https://github.com/opensearch-project/opensearch-plugins/blob/main/UPGRADING.md#rest-apis-backward-compatibility AFAIK

**Review by dblock — CHANGES_REQUESTED** [review]
This also needs tests ensuring that the legacy path works.

**Comment by eirsep** [comment]
> Is won't have legacy API to fallback to, so tests will start failing 50% of the time. I'd just write a couple of tests ensuring they hit the legacy paths for at least some use case and call it a day.

Agreed @dblock. Adding dedicated bwc tests.

File: src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java (hunk 4 above)
Comment (eirsep) [reply src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/plugin/AsynchronousSearchPlugin.java:87]: Will rename this.

File: build.gradle [hunk 5]
```diff
@@ -214,6 +214,7 @@ integTest {
     }
 }
 
+
```
Comment (Bukhtawar) [inline build.gradle:217]: nit: remove?

**Review by Bukhtawar — APPROVED** [review]
Thanks Surya. LGTM

**Pull request merged**